nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_rviz_plugins/nav2_panel.hpp"
+
+#include <dirent.h>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QPushButton>
+#include <QtConcurrent/QtConcurrent>
+#include <QVBoxLayout>
+
+#include "rviz_common/display_context.hpp"
+
+namespace nav2_rviz_plugins
+{
+
+Nav2Panel::Nav2Panel(QWidget * parent)
+: Panel(parent)
+{
+  // Create the control button and its tooltip
+
+  start_stop_button_ = new QPushButton;
+  pause_resume_button_ = new QPushButton;
+
+  // Create the state machine used to present the proper control button states in the UI
+
+  const char * startup_msg = "Configure and activate all nav2 lifecycle nodes";
+  const char * shutdown_msg1 = "Deactivate and cleanup all nav2 lifecycle nodes";
```

Comment:
Do we want to change the pause button to just deactivate, not cleanup? I think that's more consistent with a pause

Reply:
Currently, all of the parameters are read during the on_configure callback. So, to bring the system to that state again we have to deactivate and cleanup. As we get finer grained with parameters and allow some to be changed in the on_activate callback, we'll need to know which parameters are associated with which states (that is, what state must the node be in for the parameter to be updated). I think the Pause and Resume services will become obsolete at this point. For now, it's an easy way to bring all nodes to the initial state so that they can be reconfigured.